### ISSUE TITLE
AD->getUidByIsMemberOf is missing

### ISSUE THREAD

**Issue by jtgasper3** [ucidentity/Grouper-ActiveMQ-Provisioner#11]
AD->getUidByIsMemberOf does not seem to exists in UCB or CMU code, but it is referenced several (3) times in ActiveMQ.pm code.

**Comment by jtgasper3** [comment]
@jeffmc930,

I'm finding other weird things as well that seem like it may not be running as anticipated.

Did you get this code running successfully with AD?

**Comment by jeffmc930** [comment]
Yes. The code did work as it was. When I went looking for getUidByIsMemberOf, I found it in ActiveMQ.pm rather than AD.pm. I haven’t ever made any changes to that file. That particular method is used to confirm deletion of all the IsMemberOf entries related to the group that is being deleted.

Jeff

**Comment by jtgasper3** [comment]
Thanks Jeff,

I guess I should add. Were you doing?:

```
changeLog.consumer.ldapIMO.syncType=basic
```

or 

```
changeLog.consumer.ldapIMO.syncType=isMemberOf
```

**Comment by jeffmc930** [comment]
Okay. Now I see the confusing part. AD would use basic.

**Comment by jtgasper3** [comment]
OK. Thanks.

**Comment by jeffmc930** [comment]
I moved those choices from the intermediate app that came with the original setup. As I’m thinking about it now, I’m wondering if you have any thoughts about that choice? It occurs to me that another way of solving that issue would be to just have the perl code ignore messages that it can’t handle. So the 389.pm code can handle isMemberOf messages, but the AD.pm code just ignores them. The CLC instance would just put all messages on the queue.

What do you think?

**Comment by jtgasper3** [comment]
That's one way, and it would probably work fine. We can just use lots of `eval {};` calls, but include some debug statements so it is clear that the fail is expected. (I can't believe how much time I've spent tracking down `die()` statements that pop the stack 3 or 4 calls, but don't include any logging statements as to why.

Another option is something we did for another client. We modded the CLC payload for the queue (they were using AWS SQS) to include the Grouper attributes. Then the Python-based provisioner knew where to send it; one or more targets. Are all groups going to both AD and LDAP? Or some here and some there?

Am I also reading this code correctly that multiple instances need to be fired up; one for AD and one for LDAP? They both can theoretically share the same config.pl file, besides that one parameter (ldap=>env) which says which one to actually use. I'm not crazy about this pattern being used.
